core/src/main/resources/hudson/PluginManager/_table.js

```diff
@@ -10,46 +10,214 @@ function checkPluginsWithoutWarnings() {
+                                htmlDivElement.classList.add('alert', 'alert-warning');
+                                htmlDivElement.innerHTML = plugin.adoptMe
+                                htmlTableDataCellElement.append(htmlDivElement)
+                            }
+
+                            if (plugin.newerVersionAvailableNotOffered) {
+                                htmlDivElement = document.createElement('div');
+                                htmlDivElement.classList.add('alert', 'alert-info');
+                                htmlDivElement.innerHTML = plugin.newerVersionAvailableNotOffered
+                                htmlTableDataCellElement.append(htmlDivElement)
+                            }
+                        }
+
+                        function createVersionEntry() {
+                            var htmlTableDataCellElement = document.createElement('td');
+                            htmlTableDataCellElement.classList.add('pane')
+                            htmlTableDataCellElement.append(document.createTextNode(plugin.version));
+                            htmlTableRowElement.append(htmlTableDataCellElement);
+                        }
+
+                        function createReleasedAgo() {
+                            var htmlTableDataCellElement = document.createElement('td');
+                            htmlTableDataCellElement.classList.add('pane')
+                            if (plugin.releaseTimestamp) {
+                                var htmlTimeElement = document.createElement('time');
+                                htmlTimeElement.dateTime = plugin.releaseTimestamp.iso8601;
+                                htmlTimeElement.textContent = plugin.releaseTimestamp.displayValue;
+                                htmlTableDataCellElement.append(htmlTimeElement);
+                            }
+                            htmlTableRowElement.append(htmlTableDataCellElement);
+                        }
+
+                        if (pluginsTable.dataset.hasadmin === 'true') {
+                            createCheckBox();
+                        }
+                        createNameEntry();
+                        createVersionEntry();
+                        createReleasedAgo();
+
+                        tbody.append(htmlTableRowElement);
+                    })
+                })
             } else {
-                items[i].addClassName("hidden");
+                var filter = e.value.toLowerCase().trim();
+                var filterParts = filter.split(/ +/).filter(function (word) {
+                    return word.length > 0;
+                });
+                var items = document.getElementsBySelector("TR.plugin");
+                var anyVisible = false;
+                for (var i = 0; i < items.length; i++) {
+                    if ((filterParts.length < 1 || filter.length < 2) && items[i].hasClassName("hidden-by-default")) {
```

Comment:
hasClassName and addClassName seem to be prototypejs funcitons.
classList is the official one - https://developer.mozilla.org/en-US/docs/Web/API/Element/classList 

items[i].classList.contains('hidden-by-default')
items[i].classList.add('hidden')

(or even use toggle)

Reply:
Looks like you use classList higher up. Any chance to nix any prototypejs is super high priority to me

Reply:
This is the moved code (in the else block) I haven’t touched it other than disabling it on the available page.

But yes can clean this up